### PR TITLE
Cleanup stale indexes and metadata columns

### PR DESCRIPTION
**Pull request by xispa**
## Description of the issue/feature this PR addresses

**Merge #2334 first**

This Pull Request removes indexes and metadata columns that are no longer needed

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: src/senaite/core/upgrade/v02_05_000.py [hunk 1]
```diff
@@ -380,3 +383,55 @@ def remove_legacy_reports(tool):
...
+        PORTAL_CATALOG: [
+            "getClientID",
+        ]
+    }
+    columns_to_remove = {
+
+    }
+
+    portal = api.get_portal()
+    for obj in portal.objectValues():
```
Comment (ramonski) [inline src/senaite/core/upgrade/v02_05_000.py]: Make it better explicit instead of iterating over all objects in the portal, e.g.:
```python
base_catalog_idxs = ["CreationDate", ]
for cat_id, idxs in indexes_to_remove.items():
    cat = api.get_tool(cat_id)
    if isinstance(cat, BaseCatalog):
        idxs.extend(base_catalog_idxs)
    ...
```

**Review by ramonski — APPROVED** [review]

**Comment by ramonski** [comment]
This PR is marked as `Draft`. Are you still on it or can we merge already?

**Comment by ramonski** [comment]
Would you mind to resolve the conflicts of this PR and check if everything is still valid with the new catalogs?

File: src/senaite/core/browser/samples/view.py [hunk 2]
```diff
@@ -143,22 +143,9 @@ def __init__(self, context, request):
                 "toggle": True}),
             ("ClientID", {
                 "title": _("Client ID"),
-                "index": "getClientID",
```
Comment (ramonski) [inline src/senaite/core/browser/samples/view.py]: Why did you remove this Index? Without this it is no longer possible to sort the columns by client ID.

**Review by ramonski — CHANGES_REQUESTED** [review]
Would you mind to remove only those indexes:

- CreationDate
- getProvince
- getDistrict
- getReceivedBy
- getSampler

Furthermore, the Upgrade step needs to be changed to `2519`.

File: src/senaite/core/browser/samples/view.py (hunk 2 above)
Comment (xispa) [reply src/senaite/core/browser/samples/view.py]: Thanks for the heads-up. Restored with https://github.com/senaite/senaite.core/pull/2335/commits/1b6e61d23174eaa1730538a481d06852ad4d03a6

**Comment by xispa** [comment]
Overtaken by events
